src/main/java/com/coveo/spillway/storage/utils/CacheSynchronization.java

```diff
@@ -65,10 +65,14 @@ public void run() {
                 .entrySet()
                 .forEach(
                     valueEntry -> {
-                      int cost = valueEntry.getValue().getDelta();
-
                       LimitKey limitKey = valueEntry.getKey();
 
+                      if (!limitKey.isDistributed()) {
```

Comment:
Might be a better fit in a `filter()` on the entrySet ? 

Also.. I think I'm missing something: Why are some keys distributed and others not? Shouldn't it be the storage who decides if it is distributed or not?